autoload/auto_git_diff.vim

```diff
@@ -55,13 +55,15 @@ function! s:show_git_diff_impl(hash, vertsplit, opts)
 
     setlocal modifiable
 
-    redir => out
-        execute 'silent! echo system("'.prefix.diff_command.'")'
-    redir END
+    let out = system(prefix.diff_command)
+    if v:shell_error
+        echohl ErrorMsg | echom 'Could not open diff preview:'.out | echohl None
+        return
+    endif
```

Comment:
If an error has occured, the program execution returns here, and then the preview window will still be modifiable. It is not desirable.

And I suppose that the preview window should clear its contents when an error has occured because leaving contents of another commit may cause confusion.

Furthermore, in my environment, `let out = system(prefix.diff_command)` corrupt the window showing `git-rebase-todo` file when I move cursor continually. Specifing `silent!` fixes this problem. (see also: https://stackoverflow.com/questions/42780822/garbage-printed-on-mouse-move-after-set-statusline)

How about this:
```
    silent! let out = system(prefix.diff_command)

    setlocal modifiable

    if v:shell_error
        echohl ErrorMsg | echom 'Could not open diff preview:'.out | echohl None
        silent! % delete _ "clear contents
    else
        silent! % delete _
        silent! $ put=out
        silent! 1 delete _
    endif

    setlocal nomodifiable

    noremap <buffer> q :bw<cr>
    silent execute "wincmd p"
endfunction
```

Reply:
I think executing the command before opening preview window is better. I'll try it out.